library/include/rocwmma/internal/dpp_impl.hpp

```diff
@@ -103,13 +104,20 @@ namespace rocwmma
                           typename DataT>
                 ROCWMMA_DEVICE static inline DataT exec(DataT src0, DataT src1)
                 {
-                    reinterpret_cast<int32_t&>(src0) = __builtin_amdgcn_update_dpp(
-                        reinterpret_cast<int32_t const&>(src1), // fill value 'prev'
-                        reinterpret_cast<int32_t const&>(src0), // Src value
-                        DppCtrl::opCtrl(), // DPP control code
-                        WriteRowMask, // Mask for affected rows
-                        WriteBankMask, // Mask for affected banks
-                        BoundCtrl); // Fill in 0 on invalid indices
+                    constexpr size_t VecSize = sizeof(DataT) / sizeof(uint32_t);
```

Comment:
We might want to have a static_assert here that sizeof(DataT) >= sizeof(uint32_t) and that sizeof(DataT) % sizeof(uint32_t) == 0. This will prevent usage with unpacked types.

Reply:
added